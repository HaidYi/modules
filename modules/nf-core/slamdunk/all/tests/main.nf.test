nextflow_process {

    name "Test Process SLAMDUNK_ALL"
    script "../main.nf"
    process "SLAMDUNK_ALL"

    tag "modules"
    tag "modules_nfcore"
    tag "slamdunk"
    tag "slamdunk/all"
    tag "bedops/gtf2bed"

    setup {
        run("BEDOPS_GTF2BED") {
            script "../../../bedops/gtf2bed/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test_gtf' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("slamseq - fastq") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = BEDOPS_GTF2BED.out.bed
                input[3] = [[], []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.bam.collect { file(it[1]).getName() },
                    process.out.filtered_bam.collect { file(it[1]).getName() },
                    process.out.filtered_bai.collect { file(it[1]).getName() },
                    process.out.vcf,
                    process.out.tsv,
                    process.out.plus_bedgraph,
                    process.out.mins_bedgraph,
                    process.out.versions
                ).match() }
            )
        }

    }

    test("slamseq - fastq - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test_ref' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = BEDOPS_GTF2BED.out.bed
                input[3] = [[], []]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.bam.collect { file(it[1]).getName() },
                    process.out.filtered_bam.collect { file(it[1]).getName() },
                    process.out.filtered_bai.collect { file(it[1]).getName() },
                    process.out.vcf,
                    process.out.tsv,
                    process.out.plus_bedgraph,
                    process.out.mins_bedgraph,
                    process.out.versions
                ).match() }
            )
        }

    }

}
