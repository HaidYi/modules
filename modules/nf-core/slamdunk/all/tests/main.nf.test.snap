{
    "slamseq - fastq - stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_filtered.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_filtered.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_plus.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_mins.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,21819f13dfa3f8fec1d9b2c67c77143b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-26T16:43:18.409251756"
    },
    "slamseq - fastq": {
        "content": [
            [
                "test_1.fastq_slamdunk_mapped.bam"
            ],
            [
                "test_1.fastq_slamdunk_mapped_filtered.bam"
            ],
            [
                "test_1.fastq_slamdunk_mapped_filtered.bam.bai"
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1.fastq_slamdunk_mapped_filtered_snp.vcf:md5,392b713d37020f032f5a03cb2a38f8c6"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1.fastq_slamdunk_mapped_filtered_tcount.tsv:md5,dbe22b81b5eff6aa30af7ea69f266116"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1.fastq_slamdunk_mapped_filtered_tcount_plus.bedgraph:md5,4c9cd48676bb947ca8381bc45178ae30"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_1.fastq_slamdunk_mapped_filtered_tcount_mins.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,21819f13dfa3f8fec1d9b2c67c77143b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-22T16:05:21.093906123"
    }
}